packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -143,6 +154,33 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
       [isDraggable, onPointerDown],
     );
 
+    // Prevents dialog from moving when window is being resized
+    React.useLayoutEffect(() => {
```

Comment:
useLayoutEffect causes warnings when doing SSR. https://reactjs.org/link/uselayouteffect-ssr

You can conditionally call useEffect on the first render (server) and useLayoutEffect on every subsequent render (client).

I introduced a new util in #825 but that's in the v1-update branch (v2?).
https://github.com/iTwin/iTwinUI-react/blob/a30870382e312be40678e386ea28160ac54a9e6d/packages/iTwinUI-react/src/core/utils/hooks/useIsomorphicLayoutEffect.tsx#L12-L13

Currently in the main branch every component already calls `useLayoutEffect` (via `useTheme`) so maybe it's fine for this PR but I feel like we should replace all remaining instances of useLayoutEffect in the v1-update branch.

Reply:
Maybe we can leave it for v1 update with your newly introduced hook? Or we can even do it after v1-update branch is merged.